atlasdb-api/src/main/java/com/palantir/atlasdb/transaction/api/ExpectationsStatistics.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * (c) Copyright 2022 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.transaction.api;
+
+import com.google.common.collect.ImmutableMap;
+import com.palantir.atlasdb.keyvalue.api.TableReference;
+import java.util.Optional;
+import org.immutables.value.Value;
+
+@Value.Immutable
+public interface ExpectationsStatistics {
+    Optional<Long> transactionAgeMillis();
```

Comment:
This may or may not be an implementation detail - we are not guaranteed to have a value for this. By the time the user callback is ran, we might or might not be tracking the transaction (if we evict transactions for mem leak concerns). It's also just me being overly defensive maybe? Worth noting this is the only optional field because this is tracked on the level of a TransactinoManager. Actually, this may just be a flaw in how I thought of implementing things - I will try making this cleaner!